cypress/support/commands.ts

```diff
@@ -25,13 +26,38 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
 //
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
\ No newline at end of file
+
+// intercept is only caught if service worker doesn't get the request
+// first, i.e. service-worker -> intercept -> server
+Cypress.Commands.add("mock", () => {
+  // TODO: currently intercepting all routes. Should be more specific? Maybe just to localhost/app domain?
+  // add header to requests to server for server side api mocking
+  cy.intercept({ method: "GET", url: "*" }, (req) => {
+    req.headers["x-mock"] = "true";
```

Comment:
Referenced in [handlers](https://github.com/mcMickJuice/msw-mock-poc/pull/1/files#diff-bb09d52c957fd04a8ed1a9f64c0cd6da4bb6c207f92a0fa1bb81a0e957aeaf49R35)

Reply:
Nice so we can add this incrementally then to endpoints/e2e tests.